usecases/schema/remove_duplicate_props.go

```diff
@@ -0,0 +1,71 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package schema
+
+import (
+	"context"
+
+	"github.com/sirupsen/logrus"
+	"github.com/weaviate/weaviate/entities/models"
+)
+
+func (m *Manager) removeDuplicatePropsIfPresent(ctx context.Context) error {
+	for _, c := range m.state.ObjectSchema.Classes {
+		if hasDuplicateProps(c.Properties) {
+			c.Properties = m.deduplicateProps(c.Properties, c.Class)
+		}
+	}
+
+	return nil
+}
+
+func hasDuplicateProps(props []*models.Property) bool {
+	counts := map[string]int{}
+
+	for _, prop := range props {
+		counts[prop.Name]++
```

Comment:
should propname here be case sensitive or insensitive?

would be simpler to check key presence in the map instead counting all occurrences

Reply:
both good points, addressed in the latest commit.